packages/api/cms-api/src/access-log/access-log.interceptor.ts

```diff
@@ -0,0 +1,90 @@
+import { CallHandler, ExecutionContext, Inject, Injectable, Logger, NestInterceptor, Optional } from "@nestjs/common";
+import { GqlExecutionContext } from "@nestjs/graphql";
+import { CurrentUser } from "src/user-permissions/dto/current-user";
+
+import { FILTER_AUTHENTICATED_REQUEST } from "./access-log.constants";
+import { FilterRequest } from "./access-log.module";
+
+const IGNORED_PATHS = ["/dam/images/:hash/:fileId", "/dam/files/:hash/:fileId", "/dam/images/preview/:fileId", "/dam/files/preview/:fileId"];
+
+@Injectable()
+export class AccessLogInterceptor implements NestInterceptor {
+    protected readonly logger = new Logger(AccessLogInterceptor.name);
+
+    constructor(@Optional() @Inject(FILTER_AUTHENTICATED_REQUEST) private readonly filterRequest?: FilterRequest) {}
+
+    intercept(context: ExecutionContext, next: CallHandler) {
+        const requestType = context.getType().toString();
+
+        const requestData: string[] = [];
+        let ignored = false;
+
+        if (requestType === "graphql") {
+            const graphqlExecutionContext = GqlExecutionContext.create(context);
+            const graphqlContext = graphqlExecutionContext.getContext();
+
+            if (
+                this.filterRequest &&
+                this.filterRequest({
+                    user: graphqlContext.req.user,
+                    req: graphqlContext.req,
+                })
```

Comment:
Is it a good idea to disable logging when `filterRequests` returns `true`? This might be confusing, as `Array.filter` keeps the element in the array if the function returns `true`. Should we rename the function to be more clear? For instance `shouldLogRequest: () => boolean`?

Reply:
sounds like a good idea 👍🏼 

Another name could be `ignoreLogRequest: () => boolean`?
Consider the example of `shouldLogRequest`. If the function returns `true`, it might be misleading, as `true` suggests logging the request rather than ignoring it.

Reply:
> If the function returns true, it might be misleading, as true suggests logging the request rather than ignoring it.

If the function returns true, it should log the request of course. The default implementation can be `() => true`

> Another name could be ignoreLogRequest: () => boolean?

IMO this is more confusing. To log a request, I'd need to return `false`. Seems like a double negative too me

Reply:
> it might be misleading, as true suggests logging the request rather than ignoring it.

I don't think this is misleading, it is very explicit what is happening.

Following all arguments, `shouldLogRequest: () => boolean` seems the best

Reply:
> shouldLogRequest: () => boolean seems the best

agree

Reply:
In general I also like more the positive `shouldLogRequest` and even though I find the name not perfect I don't know a better one.

Reply:
done in d0f7cdb9a4582de585dc80a9b167d9d7b3a27123